core/server/master/src/main/java/alluxio/master/file/MetadataSyncLockManager.java

```diff
@@ -0,0 +1,88 @@
+package alluxio.master.file;
+
+import alluxio.AlluxioURI;
+import alluxio.collections.LockPool;
+import alluxio.concurrent.LockMode;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.resource.LockResource;
+
+import com.google.common.base.Preconditions;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Class for managing metadata　sync locking.
+ */
+public class MetadataSyncLockManager {
+  /**
+   * Pool for supplying metadata sync locks.
+   *
+   * We use weak values so that when nothing holds a reference to
```

Comment:
is this line true? AFAIR the lock pool uses strong references?

Reply:
yea this is true